src/Files.App/Services/App/AppUpdateStoreService.cs

```
@@ -180,43 +181,39 @@ public async Task CheckAndUpdateFilesLauncherAsync()
 		{
 			var destFolderPath = Path.Combine(UserDataPaths.GetDefault().LocalAppData, "Files");
 			var destExeFilePath = Path.Combine(destFolderPath, "Files.App.Launcher.exe");
+			var branchFilePath = Path.Combine(destFolderPath, "Branch.txt");
 
-			if (Path.Exists(destExeFilePath))
-			{
-				var hashEqual = false;
-				var srcHashFile = await StorageFile.GetFileFromApplicationUriAsync(new Uri("ms-appx:///Assets/FilesOpenDialog/Files.App.Launcher.exe.sha256"));
-				var destHashFilePath = Path.Combine(destFolderPath, "Files.App.Launcher.exe.sha256");
+			// If Files.App.Launcher.exe doesn't exist, no need to update it.
+			if (!File.Exists(destExeFilePath))
+				return;
 
-				if (Path.Exists(destHashFilePath))
+			// Check if the launcher file is associated with the current branch of the app.
+			if (File.Exists(branchFilePath))
+			{
+				try
 				{
-					await using var srcStream = (await srcHashFile.OpenReadAsync()).AsStream();
-					await using var destStream = File.OpenRead(destHashFilePath);
-
-					hashEqual = HashEqual(srcStream, destStream);
+					var branch = await File.ReadAllTextAsync(branchFilePath, Encoding.UTF8);
+					if (!string.Equals(branch.Trim(), "files-dev", StringComparison.OrdinalIgnoreCase))
+						return;
 				}
-
-				if (!hashEqual)
+				catch { }
```

Would it be useful to add logging to the empty catch blocks?  In case of issues, we'd have visibility into whether it's file permission problems, corruption, etc.